### ISSUE TITLE
Magit Rev mode: way to unlimit to all files

### ISSUE THREAD

**Issue by grothesque** [magit/magit#3952]
When examining code, I like to look at the history of some file.  Pressing RET on a commit in the log buffer shows a "Magit Rev mode" buffer that is limited to the file.  Now, there does not seem to be an easy way to "unlimit" the display so that the complete commit is shown.  (At least I couldn't find it.)  I think that it would be useful.

**Comment by kyleam** [comment]
You can call `magit-diff-toggle-file-filter`, by default bound to <kbd>DF</kbd>.

**Comment by grothesque** [comment]
Thanks a lot, that's exactly what I've been looking for.

So the problem is not the absence of this functionality, but that it's difficult to find.  Since the topmost line of the buffer says "limited to file ...", I searched the full text of the magit info pages for the word "limit", but that didn't lead me to `magit-diff-toggle-file-filter` since the term used there is "filter" and not "limit".

A possible solution would be to use the word "filter" instead of "limit" in the topmost line of the commit buffer.

**Comment by kyleam** [comment]
> I searched the full text of the magit info pages for the word "limit", but that didn't lead me to `magit-diff-toggle-file-filter` since the term used there is "filter" and not "limit".

True, that doesn't lend itself to a search.  Looking over the section for the revision buffer would point to the command though.

> A possible solution would be to use the word "filter" instead of "limit" in the topmost line of the commit buffer.

I think that's a good suggestion.  I'll do that (my) tonight unless there are objections in the meantime.

**Comment by grothesque** [comment]
> I think that's a good suggestion.  I'll do that (my) tonight unless
> there are objections in the meantime.

Thanks.  The only problem is how to avoid a clumsy wording.

Perhaps turn "Commit v1.2.3 limited to file src/foo.c" into
"Commit v1.2.3, filter: src/foo.c"?

**Comment by kyleam** [comment]
>  The only problem is how to avoid a clumsy wording.

Is "filtered to" that clumsy?

**Comment by jguenther** [comment]
FWIW, I've been looking for exactly this for quite a while, I worked around it but it always seemed like I must be missing something obvious. I just never happened to notice `magit-diff-toggle-file-filter`.

IMO magit should retain similar terminology to git. From the `git diff` manpage:
```
<path>...
           The <paths> parameters, when given, are used to limit the diff to the named paths (you can give directory names and get diff for all files under them).
```

I think that it still makes sense to retain the "limit" terminology--maybe `magit-diff-toggle-file-filter` should be renamed/aliased to `magit-diff-toggle-file-limit`? "Filter" does already have a different meaning for git too (e.g. `git diff --diff-filter`)

**Comment by kyleam** [comment]
@jguenther  Thanks for weighing in.

I think that's a fair point about `git-diff`s manpage using "limit" in its description and about possible confusion with `--diff-filter`.  I should've thought more carefully about using consistent phrasing in 74cc5e64 (magit-revision-toggle-file-filter: new command, 2017-05-08).  My guess for why that commit (and the discussion surrounding it) used "file filter" and "file restriction" is that, while the verb form ("limit to files") sounds fine, using it as a noun ("file limit") sounds off to me, perhaps because it makes me think of a count-based limit.

So the main options under consideration for making the command easier to discover are

  1) use "filtered to" in the revision buffer header.

     That would allow discovering `magit-diff-toggle-file-filter` through methods like `command-apropos` and searching the manual.

     Along with this, we should probably also work "limit" into the docstring and manual entry for `magit-diff-toggle-file-filter`.  That way, if someone is searching the manual for "limit" based on the word from the manpage, they will still come across the command.

  2) rename the command to `magit-diff-toggle-file-limit`, defining `magit-diff-toggle-file-filter` as an obsolete alias.

     Conveniently the <kbd>DF</kbd> binding still works if you take "F" for "file" rather than "filter".

I'm leaning toward no. 2, despite thinking "file limit" sounds odd.

**Comment by tarsius** [comment]
> using it as a noun ("file limit") sounds off to me, perhaps because it makes me think of a count-based limit.

To me too.

My preferred option would be to not make any changes.

----

I have just spend quite some time going back and forth between variations of the above and iterations that additionally argue *why* that is so.

I'll go with a third option: use this as a first draft for a boilerplate response to be used in similar situations.

----

First of all, nobody did anything wrong at all.

At the same time issues like this suck up too much time and I would like to do something like this. I'll try to summarize how that usually play out.

1. Some user opens a feature request, mentioning that they looked at the documentation and could not find anything relevant.
2. Some maintainer point out that the feature already exists and quickly summarize how to us it.
3. The user confirm that this is what they were looking for.
4. The user wants to help and explains what would have enabled them to find the relevant documentation.
5. A quick "That already exists, see ..." has just turned into serious work for the maintainer.

The user does this in order to help. After all they know about the feature now; they just want to make sure that others can find it too without having to also bother the maintainers.

The user assumes that because they have not been able to find the documentation in its current form, that other users (including themselves on another day) are likely to have the same issue.

While the user thinks that they would have found the relevant documentation if the improvements that they are making had already been incorporated in the past, they cannot really be sure of that, and neither can the maintainer.

The user may provide arguments as to why certain changes should be made. Even though these are reasonable arguments the maintainers may disagree for a variety of reasons; including reasons they are not perfectly clear about themselves. The maintainer may even disagree only very mildly. Or have no opinion at all.  That makes things harder not easier.

At least around here, this usually results in the maintainer more or less doing what the user suggested because this is the path of least resistance. Looked at in isolation this is probably the best outcome. Even if the change is not objectively better there is at least the benefit of the user feeling good about having made a difference and the maintainer not feeling bad about either having said *no* without sufficiently justifying that or else having wasting time on an issue that really just isn't worth the effort.

If this keeps happening however, then the "probably not an improvement, possibly even slight worse than before" changes accumulate.

Maybe it would be better to leave the path of least resistance and instead go with "if in doubt, then trust the author's initial intuition". Of course we shouldn't *always* do that. The maintainers will obviously sometimes overlook issues that affect many users. So they must listen to users and not assume they always know better. On the other hand just because a handful of users report that something needs to be improved that doesn't mean that any other users would benefit from the suggested change. The opposite might be the case.

**In summary; It's not always easy to know what is the best option and if the potential gain of choosing the best option is likely minimal, then we should more often simply sticking with the status quo.**

**Comment by kyleam** [comment]
@tarsius I initially had a "do nothing ..." item in the options list, but I dropped it because I wasn't able to readily formulate the rationale for it.  Thanks for stepping back and writing a general defense of that option much more clearly than I could.

**Comment by usq** [comment]
Just ran into the same issue as well (for the 10th time until I found this issue) and wanted to leave a suggestion:

From how the magit-log buffer works (RET on a commit brings me into the magic-revision buffer), RET on the revision at the top of the magic-revision buffer could show the full commit without any filter. At the moment I get "There is no thing at point that could be visited" so I assume that action is still "free".

It's just an idea, so please feel free to disregard :)
